packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -91,6 +109,36 @@ export default class Sayt extends LitElement {
     if (changedProps.has('visible')) {
       this.hidden = !this.visible;
     }
+    if (changedProps.has('searchbox')) {
+      const oldSearchbox = changedProps.get('searchbox') as string;
+      if (oldSearchbox) {
+        this.toggleSearchboxListener('remove', 'input', this.processSearchboxInput, oldSearchbox);
+      } else {
+        this.toggleSearchboxListener('remove', 'sfx::searchbox_change', this.processSfxSearchboxChange);
+      }
+      if (this.searchbox) {
+        this.toggleSearchboxListener('add', 'input', this.processSearchboxInput, this.searchbox);
+      } else {
+        this.toggleSearchboxListener('add', 'sfx::searchbox_change', this.processSfxSearchboxChange);
+      }
+    }
+  }
+
+  /**
+   * Toggle the events being registered and unregisterd when the `searchbox` property changes.
+   *
+   * @param toggle A string to indicate the type of eventListener(add or remove).
+   * @param eventName The name of the eventListener to register or unregister.
+   * @param eventCallback The name of the callback function triggered by the eventListener.
+   * @param searchboxType The `oldSearchbox` or `this.searchbox`.
+   */
+  toggleSearchboxListener(toggle: string, eventName: string, eventCallback: any, searchboxType?: string) :void {
```

Comment:
I think a better factoring is to simply pass down the searchbox ID (type `string`) and whether to add or remove (type `'add' | 'remove'`). You already know which event callback to use and under which event name based on the searchbox ID: if it's on  the searchbox itself, it's `input` and `this.processSearchboxInput`, and if it's on `window`, it's `sfx::searchbox_change` and `this.processSfxSearchboxChange`. If you do it this way, you can replace all the `if` statements in the calling code with a simple call to this function.

I also wouldn't call it "toggle" because you aren't really toggling anything in this function. Maybe `setSearchboxListener`.